src/target/riscv/riscv.c

```diff
@@ -3061,32 +3061,28 @@ static int riscv_virt2phys(struct target *target, target_addr_t virtual, target_
 			virtual, physical);
 }
 
+static int check_virt_memory_access(struct target *target, target_addr_t address,
+			uint32_t size, uint32_t count)
+{
+	const bool is_misaligned = address % size != 0;
+	// TODO: This assumes that size of each page is 4 KiB, which is not necessarily the case.
+	const bool crosses_page_boundary = RISCV_PGBASE(address + size * count - 1) != RISCV_PGBASE(address);
+	if (is_misaligned && crosses_page_boundary) {
+		LOG_TARGET_ERROR(target, "Mis-aligned memory read|write (address=0x%" TARGET_PRIxADDR ", size=%d, count=%d)"
+			" would access an element across page boundary. This is not supported.",
+			address, size, count);
+		return ERROR_FAIL;
```

Comment:
Here I meant that the actual type of the memory operation would be printed (`read` or `write`):

```suggestion
	if (is_misaligned && crosses_page_boundary) {
		LOG_TARGET_ERROR(target, "Mis-aligned memory %s (address=0x%" TARGET_PRIxADDR ", size=%d, count=%d)"
			" would access an element across page boundary. This is not supported.",
			is_write ? "write" : "read", address, size, count);
		return ERROR_FAIL;
```

Reply:
Aha, now I understand that the `Optional: The error print can also be moved one level up to riscv_rw_memory(), if you find it more convenient.` in the original comment is because `is_write `is required.
Addressed.